src/main/java/crawlercommons/mimetypes/MimeTypeDetector.java

```diff
@@ -104,62 +102,53 @@ public MimeTypeDetector() {
         mimeTypes = new ArrayList<>();
 
         // Add all text patterns without and with a BOM.
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?xml", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?XML", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<!--", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<urlset", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<sitemapindex", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<rss", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<feed", true, true));
-        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<rdf", true, true));
-
-        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "http://", true, true));
-        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "https://", true, true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?xml", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<?XML", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<!--", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<urlset", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<sitemapindex", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<rss", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<feed", true));
+        mimeTypes.add(new MimeTypeEntry(XML_MIMETYPES[0], "<rdf", true));
+
+        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "http://", true));
+        mimeTypes.add(new MimeTypeEntry(TEXT_MIMETYPES[0], "https://", true));
 
         mimeTypes.add(new MimeTypeEntry(GZIP_MIMETYPES[0], "\037\213"));
         mimeTypes.add(new MimeTypeEntry(GZIP_MIMETYPES[0], 0x1F, 0x8B));
 
         maxPatternLength = 0;
         for (MimeTypeEntry entry : mimeTypes) {
             int length = entry.getPattern().length;
-            if (entry.allowBOM)
-                length += UTF8_BOM.length;
-            if (entry.allowLeadingSpace)
+            if (entry.isTextPattern)
                 length += LEADING_WHITESPACE_MAX_SKIP;
             maxPatternLength = Math.max(maxPatternLength, length);
         }
     }
 
     public String detect(byte[] content) {
-        return detect(content, 0, content.length);
+        return detect(content, content.length);
     }
 
-    public String detect(byte[] content, int offset, int length) {
-        int offsetBOM = -1;
-        int offsetSpace = -1;
+    public String detect(byte[] content, int length) {
+        int offsetText = -1;
```

Comment:
I don't understand why this is initialized here, and changed in the per-mimetype loop below (more comments there).

Reply:
OK, I get it; you only want to do the advance once.